src/app/MarineApp/modules/MetricsWidgets/VisibleWidgets.provider.ts

```diff
@@ -0,0 +1,10 @@
+import { useEffect } from "react"
+import { notifyParams, useVisibleWidgetsStore } from "./VisibleWidgets.store"
+
+export const useVisibilityNotifier = ({ widgetName, visible }: notifyParams) => {
+  const visibleWidgetsStore = useVisibleWidgetsStore()
+
+  useEffect(() => {
+    visibleWidgetsStore.notifyVisibility({ widgetName, visible })
+  }, [visibleWidgetsStore, visible, widgetName])
```

Comment:
Including visibleWidgetsStore in the list of dependencies will likely not work, since it's an object and Javascript will think it changes on every render. You can leave it out of the dependencies and suppress the linter warning with a comment.